lib/redis/client.rb

```diff
@@ -79,11 +79,14 @@ def initialize(options = {})
 
       @pending_reads = 0
 
-      if options.include?(:sentinels)
-        @connector = Connector::Sentinel.new(@options)
-      else
-        @connector = Connector.new(@options)
-      end
+      @connector =
+        if options.include?(:sentinels)
+          Connector::Sentinel.new(@options)
+        elsif options.include?(:connector) && options[:connector].is_a?(Class)
```

Comment:
I wouldn't check for `is_a?(Class)`, `respond_to?(:new)` would be better as it would allow for facades etc.